server/JsonStreamingServer/suppliers/JsonStreamingServer.Suppliers.FileStream/Services/CsvHotelReaderService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using JsonStreamingServer.Suppliers.FileStream.Services.Interfaces;
+using JsonStreamingServer.Core.Models.Domain;
+using CsvHelper;
+using Serilog;
+using System.Globalization;
+using System.Reflection;
+
+namespace JsonStreamingServer.Suppliers.FileStream.Services
+{
+    public class CsvHotelReaderService : ICsvHotelReaderService
```

Comment:
I would only implement IDisposiable and I would keep the state/stream in this class. I would also rename it to CsvFileHotelOfferStreamReader